reference/src/layout/unions.md

```diff
@@ -29,8 +29,17 @@ largest field, and the offset of each union field within its variant. How these
 are picked depends on certain constraints like, for example, the alignment
 requirements of the fields, the `#[repr]` attribute of the `union`, etc.
 
-[padding]: ../glossary.md#padding
-[layout]: ../glossary.md#layout
+Unions may contain both [exterior][exterior padding] and [interior padding]. In the below diagram, exterior padding is marked by `EXT`, interior padding by `INT`, and bytes that are padding bytes for a particular field but not padding for union as a whole are marked `NON`:
+
+```text
+[ EXT [ field0_0_ty | INT | field0_1_ty | INT ] EXT ]
+[ EXT [ field1_0_ty | INT | NON NON NON | INT ] EXT ]
+[ EXT | NON NON NON | INT [ field2_0_ty ] INT | EXT ]
+```
```

Comment:
IMO a simpler way to define this is to say: for each 'row' of the union, we define the set of padding bytes as
- whatever padding bytes the field has
- all the bytes before and after the field

Then the padding of the union is simply the intersection of those sets of padding bytes.

Reply:
> Then the padding of the union is simply the intersection of those sets of padding bytes.

I was assuming that for a Repr-raw union, we would want the set of padding bytes to be exactly the exterior padding bytes, i.e., the bytes that are padding before and after all fields, ignoring field-internal padding. Is this not true?

Reply:
It is not true for current unions, there were examples somewhere of types like `MaybeUninit<(u16, u32)>` being passed in 2 registers and thus having the padding of the fields lost.

For a "preserve all bits union" I was assuming we have no padding at all.